src/fsharp/CompileOps.fs

```diff
@@ -826,10 +826,10 @@ let OutputPhasedErrorR (os: StringBuilder) (err: PhasedDiagnostic) (suggestNames
       | UndefinedName(_, k, id, suggestionsF) ->
           os.Append(k (DecompileOpName id.idText)) |> ignore
           if suggestNames then
-              let filtered = ErrorResolutionHints.FilterPredictions suggestionsF id.idText 
-              if List.isEmpty filtered |> not then
-                  os.Append(ErrorResolutionHints.FormatPredictions DecompileOpName filtered) |> ignore
-          
+              let buffer = ErrorResolutionHints.MaxBuffer(ErrorResolutionHints.maxSuggestions)
+              suggestionsF (ErrorResolutionHints.FilterSuggestions buffer id.idText)
```

Comment:
is this what you had in mind?

Reply:
yeah that's in the right direction, this buffer is purely for suggestions on a target `idText` so we could have the buffer take the `idText` in the constructor and allow `Insert` overload to accept just Values, with it running the distance/error checks, to tidy/abstract away even further?

Reply:
would also mean we could reuse jaro/distance calculation arrays, keep cached values of idText 🤔